tsa-core/src/main/kotlin/org/usvm/machine/interpreter/TvmDictOperationInterpreter.kt

```diff
@@ -1515,27 +1677,100 @@ class TvmDictOperationInterpreter(
             }
             return
         }
+        /*
+        To properly avoid forking, the code in `doDictGetImpl` must be clearly separated into interpreter part and
+        logical part. After making such a division, the logical part must be rewritten with the use of guards
+        and dictionary type (input dictionary or non-input dictionary).
+         */
+        val dicts = flattenReferenceIte(dictCellRef, extractAllocated = true)
+        val actions = dicts.map { (cond, dict) -> TvmStepScopeManager.ActionOnCondition({}, false, cond, dict) }
+        if (dicts.size == 1) {
+            // avoid forking in this case
+            doDictGetImpl(
+                scope,
+                this@with,
+                keyLength,
+                dicts.single().second,
+                keyKind,
+                key,
+                nullDefaultValue,
+                inst,
+                valueType,
+            )
+        } else {
+            scope.doWithConditions(actions) { dictCellRef ->
+                doDictGetImpl(this, this@with, keyLength, dictCellRef, keyKind, key, nullDefaultValue, inst, valueType)
+            }
+        }
```

Comment:
If you make `doWithConditions` with one variant (that has `true` guard), fork doesn't actually happen. It just changes step scope. So need for this